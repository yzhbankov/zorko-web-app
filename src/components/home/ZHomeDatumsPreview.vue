<template>
  <v-card>
    <v-card-title>
      <div>
        <div class="headline util-pointer js-ZHomeDatumsPreview-title"
             @click="onClick">{{ title }}</div>
        <span class="grey--text">connected to
          <a class="grey--text"
             :href="connection">
            {{ connection }}
          </a>
        </span>
        <br>
        <span>{{ description }}</span>
        <br>
        <span style="margin-right: 10px">
          <v-icon>info</v-icon>
          <a :href="info.site">
            {{ info.name }}
          </a>
        </span>
        <span class="grey--text">{{ looksCount }} looks created</span>
      </div>
    </v-card-title>
  </v-card>
</template>

<script>
export default {
  name: 'ZHomeDatumsPreview',
  props: {
    name: {
      type: String
    },
    title: {
      type: String
    },
    connection: {
      type: String
    },
    description: {
      type: String
    },
    info: {
      type: Object
    },
    looksCount: {
      type: Number
    }
  },
  methods: {
    onClick () {
      this.$emit('openDatum', {
        name: this.name
      })
    }
  }
}
</script>

<style scoped>
  .util-pointer {
    cursor: pointer;
    text-decoration: underline;
  }
</style>
