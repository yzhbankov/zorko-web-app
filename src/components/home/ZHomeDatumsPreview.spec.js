import {mount, shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import ZHomeDatumsPreview from './ZHomeDatumsPreview.vue'

describe('HomeDatumsPreview.vue', () => {
  let defaultProps

  beforeEach(() => {
    defaultProps = {
      name: 'car',
      title: 'Cars',
      connection: 'https://vega.github.io/vega-datasets/data/cars.json',
      description: 'Common repository for example datasets used by vega related projects',
      info: {
        name: 'vega-datasets',
        site: 'https://github.com/vega/vega-datasets'
      },
      looksCount: 5
    }
  })

  it('can initialize', () => {
    const wrapper = mount(ZHomeDatumsPreview, {
      propsData: defaultProps
    })

    expect(wrapper.vm).toBeTruthy()
  })

  it('should emmit `openDatum` when click on title', () => {
    const wrapper = mount(ZHomeDatumsPreview, {
      propsData: defaultProps
    })

    wrapper.find('.js-ZHomeDatumsPreview-title').trigger('click')

    expect(wrapper.emitted('openDatum')[0][0]).toEqual({
      name: defaultProps.name
    })
  })

  it('has same HTML structure', () => {
    const renderer = createRenderer()
    const wrapper = shallow(ZHomeDatumsPreview, {
      propsData: defaultProps
    })

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
