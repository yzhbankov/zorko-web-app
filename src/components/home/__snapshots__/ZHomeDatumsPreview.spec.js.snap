// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDatumsPreview.vue has same HTML structure 1`] = `
<v-card>
    <v-card-title>
        <div>
            <div class="headline util-pointer js-ZHomeDatumsPreview-title">Cars</div> <span class="grey--text">connected to
        <a href="https://vega.github.io/vega-datasets/data/cars.json" class="grey--text">
          https://vega.github.io/vega-datasets/data/cars.json
        </a></span>
            <br> <span>Common repository for example datasets used by vega related projects</span>
            <br> <span style="margin-right:10px;"><v-icon>info</v-icon> <a href="https://github.com/vega/vega-datasets">
          vega-datasets
        </a></span> <span class="grey--text">5 looks created</span></div>
    </v-card-title>
</v-card>
`;
